app/plugins.go

```diff
@@ -557,30 +557,32 @@ func (a *App) InitPlugins(pluginPath, webappPath string) {
 
 	utils.RemoveConfigListener(a.PluginConfigListenerId)
 	a.PluginConfigListenerId = utils.AddConfigListener(func(prevCfg, cfg *model.Config) {
-		if !*prevCfg.PluginSettings.Enable && *cfg.PluginSettings.Enable {
-			a.InitPlugins(pluginPath, webappPath)
-		} else if *prevCfg.PluginSettings.Enable && !*cfg.PluginSettings.Enable {
-			a.ShutDownPlugins()
-		} else if *prevCfg.PluginSettings.Enable && *cfg.PluginSettings.Enable {
-			a.ActivatePlugins()
-		}
-
 		if a.PluginEnv == nil {
 			return
 		}
 
+		if *prevCfg.PluginSettings.Enable && *cfg.PluginSettings.Enable {
+			a.ActivatePlugins()
+		}
+
 		for _, err := range a.PluginEnv.Hooks().OnConfigurationChange() {
 			l4g.Error(err.Error())
 		}
 	})
 
-	a.Srv.Router.HandleFunc("/plugins/{plugin_id:[A-Za-z0-9\\_\\-\\.]+}", a.ServePluginRequest)
-	a.Srv.Router.HandleFunc("/plugins/{plugin_id:[A-Za-z0-9\\_\\-\\.]+}/{anything:.*}", a.ServePluginRequest)
-
 	a.ActivatePlugins()
 }
 
 func (a *App) ServePluginRequest(w http.ResponseWriter, r *http.Request) {
+	if a.PluginEnv == nil || !*a.Config().PluginSettings.Enable {
+		err := model.NewAppError("ServePluginRequest", "app.plugin.disabled.app_error", nil, "Enable plugins to serve plugin requests", http.StatusNotImplemented)
+		err.Translate(utils.T)
+		l4g.Error(err.Error())
+		w.WriteHeader(err.StatusCode)
+		w.Write([]byte(err.ToJson()))
```

Comment:
Might need to set the content-type too. Not sure what it'll default to.

Reply:
Good call, I'll set it just in case